File: frontend/src/microservices/content/objects/ReviewEligibilityPlannerResponse.ts
// ReviewEligibilityPlannerResponse：content 域评论资格判断返回对象。

export type ReviewEligibilityPlannerResponse = {
  orderId: string
  orderItemId: string
  canReview: boolean
  alreadyReviewed: boolean
  reason: string | null
  resourceSummaryTitle: string
}
export const reviewEligibilityPlannerResponseFromJson = (json: string): ReviewEligibilityPlannerResponse =>
  JSON.parse(json) as ReviewEligibilityPlannerResponse
export const reviewEligibilityPlannerResponseToJson = (value: ReviewEligibilityPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/content/objects/ReviewEligibilityPlannerResponse.scala
package com.typesafe.travel.content.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ReviewEligibilityPlannerResponse(
    orderId: String,
    orderItemId: String,
    canReview: Boolean,
    alreadyReviewed: Boolean,
    reason: Option[String],
    resourceSummaryTitle: String
)
object ReviewEligibilityPlannerResponse:
  given sourceEncoder: Encoder[ReviewEligibilityPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[ReviewEligibilityPlannerResponse] = deriveDecoder
